server/datastore/mysql/apple_mdm.go

```diff
@@ -478,16 +478,24 @@ func upsertMDMAppleHostMDMInfoDB(ctx context.Context, tx sqlx.ExtContext, server
 		ON DUPLICATE KEY UPDATE server_url = VALUES(server_url)`,
 		fleet.WellKnownMDMFleet, serverURL)
 	if err != nil {
-		return ctxerr.Wrap(ctx, err, "last insert id mdm apple host")
+		return ctxerr.Wrap(ctx, err, "upsert mdm solution")
 	}
 
-	mdmID, err := result.LastInsertId()
-	if err != nil {
-		return ctxerr.Wrap(ctx, err, "last insert id mdm apple host")
+	var mdmID int64
+	if insertOnDuplicateDidInsert(result) {
+		mdmID, _ = result.LastInsertId()
+	} else {
+		err := tx.QueryRowxContext(ctx, `
+			SELECT id FROM mobile_device_management_solutions
+			WHERE name = ? AND server_url = ?`,
+			fleet.WellKnownMDMFleet, serverURL).Scan(&mdmID)
+		if err != nil {
+			return ctxerr.Wrap(ctx, err, "query mdm solution id")
+		}
```

Comment:
this is the fix for the issue I mention in the description about `ON DUPLICATE KEY`